appinfo/routes.php

```diff
@@ -25,7 +25,7 @@
 		//Retrieve Metadata of a list of files
 		[
 			'name' => 'metadata#get_metadata',
-			'url'  => '/metadata/{fileList}',
+			'url'  => '/metadata?fileList={fileList}',
```

Comment:
You had the right idea, but that's not a proper route when trying to catch GET parameters.
Re-read the doc:
https://docs.nextcloud.com/server/10/developer_manual/app/routes.html

And look at how it's implemented in Gallery:
https://github.com/nextcloud/gallery/blob/master/appinfo/routes.php

I think this important that you understand how API requests are made.

Also, run Codeception in verbose mode. That should show you what is sent in details.